### PR TITLE
Nicer elimination error messages for sort poly inductives

### LINKED ISSUE #18515: Elimination error message is not very clear for sort polymorphic inductive type
~~~coq
Inductive sum@{s|u v|} (A : Type@{s|u}) (B : Type@{s|v}) : Type@{s|max(u,v)} :=
  | inl : A -> sum A B
  | inr : B -> sum A B.
Arguments inl {A B}.
Arguments inr {A B}.

Check (fun p : sum@{Prop|_ _} True False => match p return Set with inl a => unit | inr b => bool end).
(*
Error: Incorrect elimination of "p" in the inductive type "sum":
the return type has sort "Type@{Set+1}" while it should be at some variable quality.
Elimination of an inductive object of sort Type is not allowed on a predicate in sort Type
because wrong arity.
*)
~~~
I think we should display the actual sort of p (Prop) here.

#### Coq Version

8.19+rc1

### PR DESCRIPTION
**Pull request by SkySkimmer**
We stop going through sort families as InQSort is not useful enough for our purposes.

Close #18515

### CODE REVIEW

File: vernac/himsg.ml [hunk 1]
```diff
@@ -212,60 +212,78 @@ let explain_reference_variables sigma id c =
...
+           str "while it should be " ++ str sorts ++ str ".") ++
+        fnl () ++
+        hov 0
+          (str "Elimination of an inductive object of sort " ++ str inds ++ spc() ++
+           str "is not allowed on a predicate in sort " ++ ppt ++ fnl () ++
+           str "because" ++ spc () ++
+           (* we could try to be more precise by adding "strict" qualifiers here
+              but IMO not worth making this text more complicated when we already
+              say which of SProp/Prop work just above. *)
+           str "proofs can be eliminated only to build proofs.")
```
Comment (SkySkimmer) [inline vernac/himsg.ml:267]: actually we should probably have something in the case where the inductive is SProp and the predicate is Prop

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: kernel/inductive.mli [hunk 2]
```diff
@@ -67,6 +67,10 @@ val type_of_inductive_knowing_parameters :
 val quality_leq : Sorts.Quality.t -> Sorts.Quality.t -> bool
 (** For squashing. *)
 
+type squash = SquashToSet | SquashToQuality of Sorts.Quality.t
+
+val is_squashed : mind_specif puniverses -> squash option
```
Comment (ppedrot) [inline kernel/inductive.mli:72]: You're the RM but in theory we try not to add APIs in backported PRs.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**